modules/contact/include/constraints/MechanicalContactConstraint.h

```diff
@@ -113,6 +114,12 @@ class MechanicalContactConstraint : public NodeFaceConstraint
   std::vector<unsigned int> _vars;
   std::vector<MooseVariable *> _var_objects;
 
+  /// gap offset from either secondary, primary or both
+  const bool _has_secondary_gap_offset;
+  MooseVariable * _secondary_gap_offset_var;
+  const bool _has_mapped_primary_gap_offset;
+  MooseVariable * _mapped_primary_gap_offset_var;
```

Comment:
I was going to ask you to make these `const`, but sigh, `MooseVariableFE<T>::getNodalValue` isn't marked `const` :-( Not your problem.